### ISSUE TITLE
Cannot build from source

### ISSUE THREAD

**Issue by kgoldman** [TigerVNC/tigervnc#1820]
**Describe the bug**
autoreconf -fiv fails, missing configure.ac

**To Reproduce**
1. clone the repo
2. autoreconf -fiv
3. autoreconf: error: 'configure.ac' is required

**Expected behavior**
Expected success.

**Server (please complete the following information):**
 - OS: Fedora 39
 - VNC server:TigerVNC
 - VNC server version: master branch
 - Server downloaded from: This repo
 - Server was started using: N/A

**Additional context**
It could be that the five steps before autoreconf are important. However, build_directory, source_directory, xorg_source, etc. are undocumented, so I could not follow them.  It didn't seem likely that configure.ac should come from somewhere else though.

**Comment by bphinz** [comment]
You should be using cmake to build.  Please see [BUILDING.txt](https://github.com/TigerVNC/tigervnc/blob/master/BUILDING.txt)

**Comment by kgoldman** [comment]
> You should be using cmake to build. Please see [BUILDING.txt](https://github.com/TigerVNC/tigervnc/blob/master/BUILDING.txt)

Perhaps, but the Linux instructions describe autotools.  cmake appears to be for mac.

**Comment by bphinz** [comment]
Cmake is used on all platforms and is a prequisite to building Xvnc (which does use autotools).  My guess is that you've missed some steps.  Follow the guide in BUILDING.txt and hopefully that should help.

**Comment by kgoldman** [comment]
I followed the guide, specifically section "Building the TigerVNC Server on Unix/Linux Systems", which is autotools.  No cmake.

I interpreted the section "Building the TigerVNC Viewer on Unix/Mac Systems" to mean Unix on Mac, but maybe it means 'do this first on all Unix'y systems'.  Also, I did not want to build the viewer, only the vncpasswd program. That seems more related to the server than the viewer.
